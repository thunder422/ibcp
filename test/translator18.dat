#####################################################
#  Translator Test #18: Complete Operator Coverage  #
#####################################################

A = NOT B
A = -B

A% = NOT B%
A% = -B%

A = B + C
A = B - C
A = B * C
A = B / C
A = B MOD C
A = B ^ C
A% = B \ C
A = B = C
A = B <> C
A = B < C
A = B <= C
A = B > C
A = B >= C
A = B AND C
A = B OR C
A = B EQV C
A = B IMP C
A = B XOR C

A = B% + C
A = B% - C
A = B% * C
A = B% / C
A = B% MOD C
A = B% ^ C
A% = B% \ C
A% = B% = C
A% = B% <> C
A% = B% < C
A% = B% <= C
A% = B% > C
A% = B% >= C
A% = B% AND C
A% = B% OR C
A% = B% EQV C
A% = B% IMP C
A% = B% XOR C

A = B + C%
A = B - C%
A = B * C%
A = B / C%
A = B MOD C%
A = B ^ C%
A% = B \ C%
A% = B = C%
A% = B <> C%
A% = B < C%
A% = B <= C%
A% = B > C%
A% = B >= C%
A% = B AND C%
A% = B OR C%
A% = B EQV C%
A% = B IMP C%
A% = B XOR C%

A% = B% + C%
A% = B% - C%
A% = B% * C%
A% = B% / C%
A% = B% MOD C%
A% = B% ^ C%
A% = B% \ C%
A% = B% = C%
A% = B% <> C%
A% = B% < C%
A% = B% <= C%
A% = B% > C%
A% = B% >= C%
A% = B% AND C%
A% = B% OR C%
A% = B% EQV C%
A% = B% IMP C%
A% = B% XOR C%

A$ = B$ + C$
A% = B$ = C$
A% = B$ <> C$
A% = B$ < C$
A% = B$ <= C$
A% = B$ > C$
A% = B$ >= C$

A$ = LEFT$(B$, 1) + C$
A% = LEFT$(B$, 1) = C$
A% = LEFT$(B$, 1) <> C$
A% = LEFT$(B$, 1) < C$
A% = LEFT$(B$, 1) <= C$
A% = LEFT$(B$, 1) > C$
A% = LEFT$(B$, 1) >= C$

A$ = B$ + RIGHT$(C$, 1)
A% = B$ = RIGHT$(C$, 1)
A% = B$ <> RIGHT$(C$, 1)
A% = B$ < RIGHT$(C$, 1)
A% = B$ <= RIGHT$(C$, 1)
A% = B$ > RIGHT$(C$, 1)
A% = B$ >= RIGHT$(C$, 1)

A$ = MID$(B$, 1) + MID$(C$, 1, 2)
A% = MID$(B$, 1) = MID$(C$, 1, 2)
A% = MID$(B$, 1) <> MID$(C$, 1, 2)
A% = MID$(B$, 1) < MID$(C$, 1, 2)
A% = MID$(B$, 1) <= MID$(C$, 1, 2)
A% = MID$(B$, 1) > MID$(C$, 1, 2)
A% = MID$(B$, 1) >= MID$(C$, 1, 2)
