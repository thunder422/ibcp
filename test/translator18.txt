
Input: A = NOT B
Output: A<ref> B CvtInt NOT CvtDbl Assign 

Input: A = -B
Output: A<ref> B -U Assign 

Input: A% = NOT B%
Output: A%<ref> B% NOT Assign% 

Input: A% = -B%
Output: A%<ref> B% -U% Assign% 

Input: A = B + C
Output: A<ref> B C + Assign 

Input: A = B - C
Output: A<ref> B C - Assign 

Input: A = B * C
Output: A<ref> B C * Assign 

Input: A = B / C
Output: A<ref> B C / Assign 

Input: A = B MOD C
Output: A<ref> B C MOD Assign 

Input: A = B ^ C
Output: A<ref> B C ^ Assign 

Input: A% = B \ C
Output: A%<ref> B C \ Assign% 

Input: A = B = C
Output: A<ref> B C = CvtDbl Assign 

Input: A = B <> C
Output: A<ref> B C <> CvtDbl Assign 

Input: A = B < C
Output: A<ref> B C < CvtDbl Assign 

Input: A = B <= C
Output: A<ref> B C <= CvtDbl Assign 

Input: A = B > C
Output: A<ref> B C > CvtDbl Assign 

Input: A = B >= C
Output: A<ref> B C >= CvtDbl Assign 

Input: A = B AND C
Output: A<ref> B CvtInt C CvtInt AND CvtDbl Assign 

Input: A = B OR C
Output: A<ref> B CvtInt C CvtInt OR CvtDbl Assign 

Input: A = B EQV C
Output: A<ref> B CvtInt C CvtInt EQV CvtDbl Assign 

Input: A = B IMP C
Output: A<ref> B CvtInt C CvtInt IMP CvtDbl Assign 

Input: A = B XOR C
Output: A<ref> B CvtInt C CvtInt XOR CvtDbl Assign 

Input: A = B% + C
Output: A<ref> B% C +%1 Assign 

Input: A = B% - C
Output: A<ref> B% C -%1 Assign 

Input: A = B% * C
Output: A<ref> B% C *%1 Assign 

Input: A = B% / C
Output: A<ref> B% C /%1 Assign 

Input: A = B% MOD C
Output: A<ref> B% C MOD%1 Assign 

Input: A = B% ^ C
Output: A<ref> B% C ^%1 Assign 

Input: A% = B% \ C
Output: A%<ref> B% CvtDbl C \ Assign% 

Input: A% = B% = C
Output: A%<ref> B% C =%1 Assign% 

Input: A% = B% <> C
Output: A%<ref> B% C <>%1 Assign% 

Input: A% = B% < C
Output: A%<ref> B% C <%1 Assign% 

Input: A% = B% <= C
Output: A%<ref> B% C <=%1 Assign% 

Input: A% = B% > C
Output: A%<ref> B% C >%1 Assign% 

Input: A% = B% >= C
Output: A%<ref> B% C >=%1 Assign% 

Input: A% = B% AND C
Output: A%<ref> B% C CvtInt AND Assign% 

Input: A% = B% OR C
Output: A%<ref> B% C CvtInt OR Assign% 

Input: A% = B% EQV C
Output: A%<ref> B% C CvtInt EQV Assign% 

Input: A% = B% IMP C
Output: A%<ref> B% C CvtInt IMP Assign% 

Input: A% = B% XOR C
Output: A%<ref> B% C CvtInt XOR Assign% 

Input: A = B + C%
Output: A<ref> B C% +%2 Assign 

Input: A = B - C%
Output: A<ref> B C% -%2 Assign 

Input: A = B * C%
Output: A<ref> B C% *%2 Assign 

Input: A = B / C%
Output: A<ref> B C% /%2 Assign 

Input: A = B MOD C%
Output: A<ref> B C% MOD%2 Assign 

Input: A = B ^ C%
Output: A<ref> B C% ^* Assign 

Input: A% = B \ C%
Output: A%<ref> B C% CvtDbl \ Assign% 

Input: A% = B = C%
Output: A%<ref> B C% =%2 Assign% 

Input: A% = B <> C%
Output: A%<ref> B C% <>%2 Assign% 

Input: A% = B < C%
Output: A%<ref> B C% <%2 Assign% 

Input: A% = B <= C%
Output: A%<ref> B C% <=%2 Assign% 

Input: A% = B > C%
Output: A%<ref> B C% >%2 Assign% 

Input: A% = B >= C%
Output: A%<ref> B C% >=%2 Assign% 

Input: A% = B AND C%
Output: A%<ref> B CvtInt C% AND Assign% 

Input: A% = B OR C%
Output: A%<ref> B CvtInt C% OR Assign% 

Input: A% = B EQV C%
Output: A%<ref> B CvtInt C% EQV Assign% 

Input: A% = B IMP C%
Output: A%<ref> B CvtInt C% IMP Assign% 

Input: A% = B XOR C%
Output: A%<ref> B CvtInt C% XOR Assign% 

Input: A% = B% + C%
Output: A%<ref> B% C% +% Assign% 

Input: A% = B% - C%
Output: A%<ref> B% C% -% Assign% 

Input: A% = B% * C%
Output: A%<ref> B% C% *% Assign% 

Input: A% = B% / C%
Output: A%<ref> B% C% /% Assign% 

Input: A% = B% MOD C%
Output: A%<ref> B% C% MOD% Assign% 

Input: A% = B% ^ C%
Output: A%<ref> B% C% ^% Assign% 

Input: A% = B% \ C%
Output: A%<ref> B% CvtDbl C% CvtDbl \ Assign% 

Input: A% = B% = C%
Output: A%<ref> B% C% =% Assign% 

Input: A% = B% <> C%
Output: A%<ref> B% C% <>% Assign% 

Input: A% = B% < C%
Output: A%<ref> B% C% <% Assign% 

Input: A% = B% <= C%
Output: A%<ref> B% C% <=% Assign% 

Input: A% = B% > C%
Output: A%<ref> B% C% >% Assign% 

Input: A% = B% >= C%
Output: A%<ref> B% C% >=% Assign% 

Input: A% = B% AND C%
Output: A%<ref> B% C% AND Assign% 

Input: A% = B% OR C%
Output: A%<ref> B% C% OR Assign% 

Input: A% = B% EQV C%
Output: A%<ref> B% C% EQV Assign% 

Input: A% = B% IMP C%
Output: A%<ref> B% C% IMP Assign% 

Input: A% = B% XOR C%
Output: A%<ref> B% C% XOR Assign% 

Input: A$ = B$ + C$
Output: A$<ref> B$ C$ +$ Assign$ 

Input: A% = B$ = C$
Output: A%<ref> B$ C$ =$ Assign% 

Input: A% = B$ <> C$
Output: A%<ref> B$ C$ <>$ Assign% 

Input: A% = B$ < C$
Output: A%<ref> B$ C$ <$ Assign% 

Input: A% = B$ <= C$
Output: A%<ref> B$ C$ <=$ Assign% 

Input: A% = B$ > C$
Output: A%<ref> B$ C$ >$ Assign% 

Input: A% = B$ >= C$
Output: A%<ref> B$ C$ >=$ Assign% 

Input: A$ = LEFT$(B$, 1) + C$
Output: A$<ref> B$ 1% LEFT$( C$ +$ Assign$ 

Input: A% = LEFT$(B$, 1) = C$
Output: A%<ref> B$ 1% LEFT$( C$ =$ Assign% 

Input: A% = LEFT$(B$, 1) <> C$
Output: A%<ref> B$ 1% LEFT$( C$ <>$ Assign% 

Input: A% = LEFT$(B$, 1) < C$
Output: A%<ref> B$ 1% LEFT$( C$ <$ Assign% 

Input: A% = LEFT$(B$, 1) <= C$
Output: A%<ref> B$ 1% LEFT$( C$ <=$ Assign% 

Input: A% = LEFT$(B$, 1) > C$
Output: A%<ref> B$ 1% LEFT$( C$ >$ Assign% 

Input: A% = LEFT$(B$, 1) >= C$
Output: A%<ref> B$ 1% LEFT$( C$ >=$ Assign% 

Input: A$ = B$ + RIGHT$(C$, 1)
Output: A$<ref> B$ C$ 1% RIGHT$( +$ Assign$ 

Input: A% = B$ = RIGHT$(C$, 1)
Output: A%<ref> B$ C$ 1% RIGHT$( =$ Assign% 

Input: A% = B$ <> RIGHT$(C$, 1)
Output: A%<ref> B$ C$ 1% RIGHT$( <>$ Assign% 

Input: A% = B$ < RIGHT$(C$, 1)
Output: A%<ref> B$ C$ 1% RIGHT$( <$ Assign% 

Input: A% = B$ <= RIGHT$(C$, 1)
Output: A%<ref> B$ C$ 1% RIGHT$( <=$ Assign% 

Input: A% = B$ > RIGHT$(C$, 1)
Output: A%<ref> B$ C$ 1% RIGHT$( >$ Assign% 

Input: A% = B$ >= RIGHT$(C$, 1)
Output: A%<ref> B$ C$ 1% RIGHT$( >=$ Assign% 

Input: A$ = MID$(B$, 1) + MID$(C$, 1, 2)
Output: A$<ref> B$ 1% MID$(2 C$ 1% 2% MID$(3 +$ Assign$ 

Input: A% = MID$(B$, 1) = MID$(C$, 1, 2)
Output: A%<ref> B$ 1% MID$(2 C$ 1% 2% MID$(3 =$ Assign% 

Input: A% = MID$(B$, 1) <> MID$(C$, 1, 2)
Output: A%<ref> B$ 1% MID$(2 C$ 1% 2% MID$(3 <>$ Assign% 

Input: A% = MID$(B$, 1) < MID$(C$, 1, 2)
Output: A%<ref> B$ 1% MID$(2 C$ 1% 2% MID$(3 <$ Assign% 

Input: A% = MID$(B$, 1) <= MID$(C$, 1, 2)
Output: A%<ref> B$ 1% MID$(2 C$ 1% 2% MID$(3 <=$ Assign% 

Input: A% = MID$(B$, 1) > MID$(C$, 1, 2)
Output: A%<ref> B$ 1% MID$(2 C$ 1% 2% MID$(3 >$ Assign% 

Input: A% = MID$(B$, 1) >= MID$(C$, 1, 2)
Output: A%<ref> B$ 1% MID$(2 C$ 1% 2% MID$(3 >=$ Assign% 

