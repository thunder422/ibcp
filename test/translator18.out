
Input: A = NOT B
Output: A = NOT B 

Input: A = -B
Output: A = -B 

Input: A% = NOT B%
Output: A% = NOT B% 

Input: A% = -B%
Output: A% = -B% 

Input: A = B + C
Output: A = B + C 

Input: A = B - C
Output: A = B - C 

Input: A = B * C
Output: A = B * C 

Input: A = B / C
Output: A = B / C 

Input: A = B MOD C
Output: A = B MOD C 

Input: A = B ^ C
Output: A = B ^ C 

Input: A% = B \ C
Output: A% = B \ C 

Input: A = B = C
Output: A = B = C 

Input: A = B <> C
Output: A = B <> C 

Input: A = B < C
Output: A = B < C 

Input: A = B <= C
Output: A = B <= C 

Input: A = B > C
Output: A = B > C 

Input: A = B >= C
Output: A = B >= C 

Input: A = B AND C
Output: A = B AND C 

Input: A = B OR C
Output: A = B OR C 

Input: A = B EQV C
Output: A = B EQV C 

Input: A = B IMP C
Output: A = B IMP C 

Input: A = B XOR C
Output: A = B XOR C 

Input: A = B% + C
Output: A = B% + C 

Input: A = B% - C
Output: A = B% - C 

Input: A = B% * C
Output: A = B% * C 

Input: A = B% / C
Output: A = B% / C 

Input: A = B% MOD C
Output: A = B% MOD C 

Input: A = B% ^ C
Output: A = B% ^ C 

Input: A% = B% \ C
Output: A% = B% \ C 

Input: A% = B% = C
Output: A% = B% = C 

Input: A% = B% <> C
Output: A% = B% <> C 

Input: A% = B% < C
Output: A% = B% < C 

Input: A% = B% <= C
Output: A% = B% <= C 

Input: A% = B% > C
Output: A% = B% > C 

Input: A% = B% >= C
Output: A% = B% >= C 

Input: A% = B% AND C
Output: A% = B% AND C 

Input: A% = B% OR C
Output: A% = B% OR C 

Input: A% = B% EQV C
Output: A% = B% EQV C 

Input: A% = B% IMP C
Output: A% = B% IMP C 

Input: A% = B% XOR C
Output: A% = B% XOR C 

Input: A = B + C%
Output: A = B + C% 

Input: A = B - C%
Output: A = B - C% 

Input: A = B * C%
Output: A = B * C% 

Input: A = B / C%
Output: A = B / C% 

Input: A = B MOD C%
Output: A = B MOD C% 

Input: A = B ^ C%
Output: A = B ^ C% 

Input: A% = B \ C%
Output: A% = B \ C% 

Input: A% = B = C%
Output: A% = B = C% 

Input: A% = B <> C%
Output: A% = B <> C% 

Input: A% = B < C%
Output: A% = B < C% 

Input: A% = B <= C%
Output: A% = B <= C% 

Input: A% = B > C%
Output: A% = B > C% 

Input: A% = B >= C%
Output: A% = B >= C% 

Input: A% = B AND C%
Output: A% = B AND C% 

Input: A% = B OR C%
Output: A% = B OR C% 

Input: A% = B EQV C%
Output: A% = B EQV C% 

Input: A% = B IMP C%
Output: A% = B IMP C% 

Input: A% = B XOR C%
Output: A% = B XOR C% 

Input: A% = B% + C%
Output: A% = B% + C% 

Input: A% = B% - C%
Output: A% = B% - C% 

Input: A% = B% * C%
Output: A% = B% * C% 

Input: A% = B% / C%
Output: A% = B% / C% 

Input: A% = B% MOD C%
Output: A% = B% MOD C% 

Input: A% = B% ^ C%
Output: A% = B% ^ C% 

Input: A% = B% \ C%
Output: A% = B% \ C% 

Input: A% = B% = C%
Output: A% = B% = C% 

Input: A% = B% <> C%
Output: A% = B% <> C% 

Input: A% = B% < C%
Output: A% = B% < C% 

Input: A% = B% <= C%
Output: A% = B% <= C% 

Input: A% = B% > C%
Output: A% = B% > C% 

Input: A% = B% >= C%
Output: A% = B% >= C% 

Input: A% = B% AND C%
Output: A% = B% AND C% 

Input: A% = B% OR C%
Output: A% = B% OR C% 

Input: A% = B% EQV C%
Output: A% = B% EQV C% 

Input: A% = B% IMP C%
Output: A% = B% IMP C% 

Input: A% = B% XOR C%
Output: A% = B% XOR C% 

Input: A$ = B$ + C$
Output: A$ = B$ + C$ 

Input: A% = B$ = C$
Output: A% = B$ = C$ 

Input: A% = B$ <> C$
Output: A% = B$ <> C$ 

Input: A% = B$ < C$
Output: A% = B$ < C$ 

Input: A% = B$ <= C$
Output: A% = B$ <= C$ 

Input: A% = B$ > C$
Output: A% = B$ > C$ 

Input: A% = B$ >= C$
Output: A% = B$ >= C$ 

Input: A$ = LEFT$(B$, 1) + C$
Output: A$ = LEFT$(B$, 1) + C$ 

Input: A% = LEFT$(B$, 1) = C$
Output: A% = LEFT$(B$, 1) = C$ 

Input: A% = LEFT$(B$, 1) <> C$
Output: A% = LEFT$(B$, 1) <> C$ 

Input: A% = LEFT$(B$, 1) < C$
Output: A% = LEFT$(B$, 1) < C$ 

Input: A% = LEFT$(B$, 1) <= C$
Output: A% = LEFT$(B$, 1) <= C$ 

Input: A% = LEFT$(B$, 1) > C$
Output: A% = LEFT$(B$, 1) > C$ 

Input: A% = LEFT$(B$, 1) >= C$
Output: A% = LEFT$(B$, 1) >= C$ 

Input: A$ = B$ + RIGHT$(C$, 1)
Output: A$ = B$ + RIGHT$(C$, 1) 

Input: A% = B$ = RIGHT$(C$, 1)
Output: A% = B$ = RIGHT$(C$, 1) 

Input: A% = B$ <> RIGHT$(C$, 1)
Output: A% = B$ <> RIGHT$(C$, 1) 

Input: A% = B$ < RIGHT$(C$, 1)
Output: A% = B$ < RIGHT$(C$, 1) 

Input: A% = B$ <= RIGHT$(C$, 1)
Output: A% = B$ <= RIGHT$(C$, 1) 

Input: A% = B$ > RIGHT$(C$, 1)
Output: A% = B$ > RIGHT$(C$, 1) 

Input: A% = B$ >= RIGHT$(C$, 1)
Output: A% = B$ >= RIGHT$(C$, 1) 

Input: A$ = MID$(B$, 1) + MID$(C$, 1, 2)
Output: A$ = MID$(B$, 1) + MID$(C$, 1, 2) 

Input: A% = MID$(B$, 1) = MID$(C$, 1, 2)
Output: A% = MID$(B$, 1) = MID$(C$, 1, 2) 

Input: A% = MID$(B$, 1) <> MID$(C$, 1, 2)
Output: A% = MID$(B$, 1) <> MID$(C$, 1, 2) 

Input: A% = MID$(B$, 1) < MID$(C$, 1, 2)
Output: A% = MID$(B$, 1) < MID$(C$, 1, 2) 

Input: A% = MID$(B$, 1) <= MID$(C$, 1, 2)
Output: A% = MID$(B$, 1) <= MID$(C$, 1, 2) 

Input: A% = MID$(B$, 1) > MID$(C$, 1, 2)
Output: A% = MID$(B$, 1) > MID$(C$, 1, 2) 

Input: A% = MID$(B$, 1) >= MID$(C$, 1, 2)
Output: A% = MID$(B$, 1) >= MID$(C$, 1, 2) 

